analyzers/src/SonarAnalyzer.CSharp/Rules/LossOfFractionInDivision.cs

```diff
@@ -106,7 +110,31 @@ private static bool TryGetTypeFromAssignmentToFloatType(SyntaxNode division, Sem
                 return type.IsAny(KnownType.NonIntegralNumbers);
             }
 
-            if (division.Parent.Parent.Parent is VariableDeclarationSyntax variableDecl)
+            var outerTuple = GetMostOuterTuple(division);
+            if (outerTuple is { Parent: AssignmentExpressionSyntax assignmentSyntax }
+                && assignmentSyntax.MapAssignmentArguments() is { } assignmentMappings)
+            {
+                var assignementLeft = assignmentMappings.Where(x => x.Right.Equals(division)).FirstOrDefault().Left;
+                if (assignementLeft != null)
+                {
+                    type = semanticModel.GetTypeInfo(assignementLeft).Type;
+                    return type.IsAny(KnownType.NonIntegralNumbers);
+                }
+            }
+            else if (outerTuple is { Parent: EqualsValueClauseSyntax { Parent: VariableDeclaratorSyntax { Parent: VariableDeclarationSyntax variableDeclaration} } })
+            {
+                var tupleArguments = ((TupleExpressionSyntaxWrapper)outerTuple).AllArguments();
+                var declarationType = semanticModel.GetTypeInfo(variableDeclaration.Type).Type;
+                var flattenTupleTypes = AllTupleElements(declarationType);
+                var argumentIndex = DivisionArgumentIndex(tupleArguments, division);
+                if (argumentIndex != -1)
+                {
+                    type = flattenTupleTypes[argumentIndex];
+                    return type.IsAny(KnownType.NonIntegralNumbers);
+                }
```

Comment:
Unfortunately, this results in an FP in cases where the indexes match, but the structure is different:

 `(int, (int, int)) t = (1, 1/3, 3);`.

While this doesn't compile, there are probably cases where this logic can be tricked with something like this (not tested by me):
```cs
(int, ValueTuple<int, int>) t = (1, (1/3, 3));
```

Reply:
Indeed, `(int, ValueTuple<int, int>) t = (1, (1/3, 3));` I documented it and added a trello card for this.